src/librbd/ImageCtx.cc

```diff
@@ -809,6 +811,20 @@ class ThreadPoolSingleton : public ThreadPool {
     }
   }
 
+  void ImageCtx::clear_pending_completions() {
+    Mutex::Locker l(completed_reqs_lock);
+    ldout(cct, 10) << "clear pending AioCompletion: count="
+                   << completed_reqs.size() << dendl;
+
+    for (xlist<AioCompletion*>::iterator it = completed_reqs.begin();
+         !it.end(); ++it) {
+      ldout(cct, 10) << "clear pending AioCompletion: " << *it << dendl;
+      (*it)->set_event_notify(false);
```

Comment:
Is this loop necessary? If the completions are already on this list, setting the flag to false doesn't help.  The only thing it does help is the destructor, but clearing the list below breaks the linkage.